quickwit-search/src/collector.rs

```diff
@@ -525,4 +549,31 @@ mod tests {
             vec![make_hit_given_split_id(1), make_hit_given_split_id(2)]
         );
     }
+
+    prop_compose! {
+        // Turns out, zero's and negative zero's u64 representation is not same.
+        // It is not relevant for our use case. For simplicity we filter the negative
+        // zero.
+        fn any_f32_without_negative_zero()(val in any::<f32>().prop_filter("Value can't be negative zero", |val| *val != -0.0)) -> f32 {
+            val
+        }
+    }
+
+    proptest! {
+        #![proptest_config(ProptestConfig::with_cases(10000))]
+        #[test]
+        fn test_proptest_f32_to_u64_compare_positive(a in 0.0..f32::MAX, b in 0.0..f32::MAX) {
```

Comment:
Why do we need the first two tests? The last one seems to cover the same search space.

Reply:
You are right. I added the third one last and did not remove the first two. It is removed now.